Scoop/Controller/Profile/EditProfileViewController.swift

```diff
@@ -0,0 +1,625 @@
+        
+        profileImageView.layer.cornerRadius = profileImageView.frame.size.width / 2
+        uploadPhotoButton.layer.cornerRadius = uploadPhotoButton.frame.size.width / 2
+    }
+    
+    // TODO: Temporary while save button is not implemented
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        updateUser()
+    }
+    
+    // MARK: - Initializers
+    
+    init(user: BaseUser, hometown: String, talkative: Float, music: Float, snack: String, song: String, stop: String) {
+        super.init(nibName: nil, bundle: nil)
+        self.user = user
+        
+        if let profilePicUrl = user.profilePicUrl {
+            profileImageView.sd_setImage(with: URL(string: profilePicUrl))
+        } else {
+            profileImageView.image = UIImage.emptyImage
+        }
+        
+        nameTextField.text = "\(user.firstName) \(user.lastName)"
+        pronounsTextField.text = user.pronouns
+        hometownTextField.text = hometown
+        classTextField.text = user.grade
+        phoneNumTextField.text = user.phoneNumber
+        talkativeSlider.setValue(talkative, animated: false)
+        musicSlider.setValue(music, animated: false)
+        snackTextField.text = snack
+        songTextField.text = song
+        stopTextField.text = stop
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Setup View Functions
+    
+    private func setupHeader() {
+        let dottedLineMultiplier = 0.52
+        let solidLineVerticalInset = -12.75
+        let solidLineMultiplier = 0.32
+        let screenSize = UIScreen.main.bounds
+        let dottedline = UIImageView(image: UIImage.dottedLine)
+        let solidline = UIView()
+        
+        dottedline.contentMode = .scaleAspectFit
+        view.addSubview(dottedline)
+        
+        dottedline.snp.makeConstraints { make in
+            make.height.equalTo(1)
+            make.width.equalTo(screenSize.width * dottedLineMultiplier)
+            make.trailing.equalToSuperview().inset(10)
+            make.top.equalTo(view.safeAreaLayoutGuide)
+        }
+        
+        solidline.backgroundColor = UIColor.black
+        view.addSubview(solidline)
+        
+        solidline.snp.makeConstraints { make in
+            make.height.equalTo(1)
+            make.width.equalTo(screenSize.width * solidLineMultiplier)
+            make.trailing.equalToSuperview()
+            make.top.equalTo(view.safeAreaLayoutGuide).inset(solidLineVerticalInset)
+        }
+        
+        navigationController?.navigationBar.prefersLargeTitles = false
+        navigationItem.title = "Edit Profile"
+        navigationController?.navigationBar.titleTextAttributes = [.font: UIFont.flowHeader]
+    }
+    
+    private func setupScrollView() {
+        scrollView.contentSize = CGSizeMake(mainStackView.frame.width, mainStackView.frame.height)
```

Comment:
Its always best to make the scrollView content size dependent on all subsequent subviews rather than setting the size to one of the contents. You can make this size dynamic by looping through all subviews and take the union of all UIView frames:
```
let contentRect: CGRect = scrollView.subviews.reduce(into: .zero) { rect, view in
    rect = rect.union(view.frame)
}
        
scrollView.contentSize = CGSize(width: view.frame.width, height: contentRect.height)
```

This way, you can continue to add views to the scrollView without having to worry about editing the contentSize. Keep in mind this needs to be called after the subviews are laid out

Reply:
What I did was arrange all the subviews inside `mainStackView`, so it's the only subview of `scrollView`. Stack views don't have a set height so the height would already be dependent on the subviews within, so if I ever added more to the view it would be a subview of the stack view and I wouldn't have to change the scroll view's `contentSize`. Would that be okay or should I still change it?

Reply:
Ig in this instance this would be okay since everything on the scrollable view is aligned on the left and right, but its always safer to constrain based on each individual subview's frame, since its not always the case where you have everything aligned